### DISCUSSION TITLE
Factory-generated Carbon Object with Mutators Returns Incorrect Attribute Value Until Model is Refreshed

### DISCUSSION THREAD

**Discussion by green-mike** [laravel/framework#52200] [Q&A]
### Laravel Version

11.15.0

### PHP Version

8.3.1

### Database Driver & Version

mysql  Ver 8.0.35-0ubuntu0.22.04.1 for Linux on aarch64 ((Ubuntu))

### Issue Description

When using a factory to generate a Carbon object and using Mutators, the access attribute retrieves the Carbon object set by the factory instead of the new value set by the mutator. The value only updates correctly after a refresh from the database.

### Steps to Reproduce

1. Create a model with a date attribute and define a mutator for this attribute.
2. Use a factory to generate an instance of this model and set the date attribute with a Carbon object.
3. Access the date attribute on the model instance.

### Expected Behavior

The date attribute should return the value set by the mutator.

### Actual Behavior

The date attribute returns the Carbon object set by the factory. The value only updates correctly after refreshing the model from the database.

### Analysis

Upon investigation, it appears that the `mutateAttributeMarkedAttribute` function directly uses the value in the `attributeCastCache`, which retains the initial Carbon object set by the factory.

https://github.com/laravel/framework/blob/1225a2ff8c8eb85240ebd8b1eb5b7442eea335c5/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L696-L700

### Steps To Reproduce

repo: https://github.com/green-mike/laravel-issue
- Create a model with an end_at attribute and define a mutator to set end_at to 23:59:59 of the current day.
- Use a factory to generate an instance of this model and set the end_at attribute with a Carbon object.
- Access the end_at attribute on the model instance.

**Comment by DanteB918** [comment]
Hello Mike, I tried out your testing environment, it looks to me it's because your `end_at` is a string in the database, but a carbon object in the factory.

in your `endAt` method, you can switch the `return (date('Y-m-d 23:59:59', strtotime($value)) ?? null);` to `return $value->endOfDay();`

and you will see the carbon object modified correctly. I'd recommend adding `protected $casts = ['end_at' => 'datetime'];` as a property to your model so that it is always a carbon object when being retrieved from the database. Or, using `now()->toDateString()` in your factory and leaving your existing code as it is.
